File: apps/api/src/domain/inventory-repository.ts
import type {
  DashboardSummary,
  ForecastResult,
  ProductFilters,
  ProductList,
  SyncRunList,
} from './models.js';

export interface InventoryRepository {
  healthCheck(): Promise<boolean>;
  getDashboardSummary(): Promise<DashboardSummary>;
  listProducts(filters: ProductFilters): Promise<ProductList>;
  getForecast(horizon: 7 | 30 | 90, productId?: string): Promise<ForecastResult>;
  listSyncRuns(): Promise<SyncRunList>;
}

File: apps/mobile/lib/domain/inventory_repository.dart
import 'models.dart';

abstract interface class InventoryRepository {
  Future<DashboardSummary> getDashboardSummary();

  Future<ForecastResult> getForecast({required int horizon});
  Future<DatasetInfo> getCurrentDataset();

  Future<List<ProductSummary>> getProducts({String? search, StockRisk? risk});

  void close();
}
